orchagent/dtelorch.cpp

```diff
@@ -0,0 +1,1384 @@
+                    SWSS_LOG_ERROR("DTEL ERROR: Failed to reset latency sensitivity");
+		    goto dtel_table_continue;
+                }
+            }
+            else if (table_attr == SINK_PORT_LIST)
+            {
+                sinkPortList.clear();
+                status = updateSinkPortList();
+                if (status != SAI_STATUS_SUCCESS)
+                {
+                    SWSS_LOG_ERROR("DTEL ERROR: Failed to reset sink port list");
+		            goto dtel_table_continue;
+                }
+            }
+            else if (table_attr == INT_L4_DSCP)
+            {
+                attr.id = SAI_DTEL_ATTR_INT_L4_DSCP;
+                attr.value.aclfield.data.u8 = 0;
+                attr.value.aclfield.mask.u8 = 0;
+                status = sai_dtel_api->set_dtel_attribute(dtelId, &attr);
+                if (status != SAI_STATUS_SUCCESS)
+                {
+                    SWSS_LOG_ERROR("DTEL ERROR: Failed to reset INT L4 DSCP value/mask");
+		    goto dtel_table_continue;
+                }
+            }
+        }
+
+dtel_table_continue:
+        it = consumer.m_toSync.erase(it);
+    }
+}
+
+bool DTelOrch::deleteReportSession(const string &report_session_id)
+{
+    sai_object_id_t report_session_oid;
+    sai_status_t status;
+
+    if (!reportSessionExists(report_session_id))
+    {
+        SWSS_LOG_ERROR("DTEL ERROR: Report session %s does not exist", report_session_id.c_str());
+        return false;
+    }
+
+    if (!getReportSessionOid(report_session_id, report_session_oid))
+    {
+        SWSS_LOG_ERROR("DTEL ERROR: Could not get report session oid for session %s", report_session_id.c_str());
+        return false;
+    }
+
+    status = sai_dtel_api->remove_dtel_report_session(report_session_oid);
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_ERROR("DTEL ERROR: Failed to delete report session %s", report_session_id.c_str());
+        return false;
+    }
+
+    m_dTelReportSessionTable.erase(report_session_id);
+    return true;
+}
+
+void DTelOrch::doDtelReportSessionTableTask(Consumer &consumer)
+{
+    SWSS_LOG_ENTER();
+
+    sai_status_t status;
+
+    auto it = consumer.m_toSync.begin();
+    while (it != consumer.m_toSync.end())
+    {
+        sai_object_id_t report_session_oid;
+
+        KeyOpFieldsValuesTuple t = it->second;
+        string report_session_id = kfvKey(t);
+
+        string op = kfvOp(t);
+
+        if (op == SET_COMMAND)
+        {
+            vector<sai_attribute_t> report_session_attr;
+            vector<sai_ip_address_t> dst_ip_list;
+            sai_ip_address_t dst_ip;
+            sai_attribute_t rs_attr;
+
+	    /* If report session already exists, delete it first */
+	    if (reportSessionExists(report_session_id))
+	    {
+	        if (!deleteReportSession(report_session_id))
+                {
+                    goto report_session_table_continue;
+                }
+	    }
+
+            for (auto i : kfvFieldsValues(t))
```

Comment:
minor issue. The indentation here doesn't look consistent

Reply:
Fixed